### ISSUE TITLE
"dep:dependency" features are not running with `--feature-powerset` and without `--optional-deps`

### ISSUE THREAD

**Issue by TimDiekmann** [taiki-e/cargo-hack#153]
I have a `Cargo.toml` which basically contains
```toml
[dependencies]
crate_a = { version = "1.33.7", optional = true }
crate_b = { version = "1.33.8" }
crate_c = { version = "1.33.9", optional = true }

[features]
crate_a = ["dep:crate_a", "crate_b/crate_a"]
great_feature = ["dep:crate_c"]
```

- `--feature-powerset` will not run for `--features crate_a`
- `--feature-powerset --optional-deps`, works with `crate_a` but also tries to run `crate_c` which fails because a feature with "dep:crate_c" is specified

Do you see this being fixed somehow? I try to avoid specifying the optional dependencies myself. The optimal solution would probably be that `--feature-powerset` will "simply" pick up `crate-a` as well as it's defined int he `[features]` section.

Without looking at the code I guess that you read all features and exclude them if `--optional-deps` is not specified and the crate is also an optional dependency?

**Comment by taiki-e** [comment]
Thanks for the report!

> Without looking at the code I guess that you read all features and exclude them if `--optional-deps` is not specified and the crate is also an optional dependency?

Yes. We need to fix this filter:

https://github.com/taiki-e/cargo-hack/blob/202e6e59d491c9202ce148c9ef423853267226db/src/features.rs#L26-L30

(This incomplete filter was added to fix breakage after namespaced and weak dependency features stabilization changed the `cargo metadata` and [caused breakage in some projects](https://github.com/taiki-e/cargo-hack/issues/145).)

**Comment by taiki-e** [comment]
Fixed in 0.5.15 (#154)
